libs/CovidDatasets.py

```diff
@@ -196,14 +196,15 @@ def combine_state_county_data(self, country, state):
             [self.DATE_FIELD, self.COUNTRY_FIELD, self.STATE_FIELD], as_index=False
         )[[self.CASE_FIELD, self.DEATH_FIELD, self.RECOVERED_FIELD]].sum()
         # Now we fill in whatever gaps we can in the state data using the county data
-        curr_date = state_data[self.DATE_FIELD].max()  # Start on the last date of state data we have
+        curr_date = max(state_data[self.DATE_FIELD].max(), county_data[self.DATE_FIELD].max()) # Start on the last date of state data we have
         county_data_to_insert = []
         while curr_date > self._START_DATE:
             curr_date -= datetime.timedelta(days=1)
             # If there is no state data for a day, we need to get some country data for the day
             if len(state_data[state_data[self.DATE_FIELD] == curr_date]) == 0:
                 county_data_for_date = copy(county_data[county_data[self.DATE_FIELD] == curr_date])
                 if len(county_data_for_date) == 0:  # If there's no county data, we're SOL.
+                    logging.info("NO COUNTY DATA: {}".format(curr_date))
```

Comment:
I think we should take this one step further and raise an exception. If we reach this point in execution, that means that for a certain date, there is no state data and no county data. That's a serious problem, which _will_ crash the model later on in execution, 100% of the time.

The exception should also indicate the date for which there is no state or county data

Reply:
I don't want to block merging on this, but I'll open an issue for how we handle this concern. I'm leaning toward not throwing an exception and having the consumer of the data be responsible for ensuring it has the necessary data points to operate.

Reply:
The issue: https://github.com/covid-projections/covid-data-model/issues/35